### PR TITLE
Add Kubernetes deployment target support

### PR DESCRIPTION
**Pull request by zachomedia**
This, along with https://github.com/OctopusDeploy/go-octopusdeploy/pull/17 (which is a dependency for these changes) allows the configuration of Kubernetes deployment targets using terraform.

### CODE REVIEW

**Comment by jeff-french** [comment]
LGTM but let's let @droyad take a look as well.

**Comment by jeff-french** [comment]
@zachomedia Bump the go-octopusdeploy dep to v1.4.0 and this will be ready to merge.

**Comment by zachomedia** [comment]
@jeff-french I've updated go-octopusdeploy to v1.4.0 after rebasing against the latest master to prevent a merge conflict.

**Comment by jeff-french** [comment]
:shipit:

File: octopusdeploy/resource_account.go [hunk 1]
```diff
@@ -48,13 +55,17 @@ func resourceAccount() *schema.Resource {
 			},
 			"tenanted_deployment_participation": {
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
```
Comment (droyad) [inline octopusdeploy/resource_account.go]: Technically this is not optional, it is likely filled in by the API. We should default this to the right value like we do with other such values to be consistent

File: octopusdeploy/resource_machine.go [hunk 2]
```diff
@@ -41,6 +42,7 @@ func resourceMachine() *schema.Resource {
 								"AzureWebApp",
 								"Ftp",
 								"AzureCloudService",
+								"Kubernetes"
```
Comment (droyad) [inline octopusdeploy/resource_machine.go]: Before 1.0 I think we should make these first class. i.e. have a `machine_kuberneties` element instead

**Pull request merged**